static/js/drafts.js

```diff
@@ -374,12 +366,35 @@ function row_with_focus() {
 
 function row_before_focus() {
     const $focused_row = row_with_focus();
-    return $focused_row.prev(".draft-row:visible");
+    const $prev_row = $focused_row.prev(".draft-row:visible");
+    // The draft modal can have two sub-sections. This handles the edge case
+    // when the user moves from the second "Other drafts" section to the first
+    // section which contains drafts from a particular narrow.
+    if (
+        $prev_row.length === 0 &&
+        $focused_row.parent().attr("id") === "other-drafts" &&
+        $("#drafts-from-conversation").is(":visible")
+    ) {
+        return $($("#drafts-from-conversation").children(".draft-row:visible").last());
+    }
+
+    return $prev_row;
```

Comment:
These changes (and the `row_after` below) together ensure that the keyboard navigation up/down works correctly.